src/tests/blueprint/t_blueprint_mpi_mesh_load_balance_related_boundary.cpp

```diff
@@ -0,0 +1,556 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and other Conduit
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Conduit.
+
+//-----------------------------------------------------------------------------
+///
+/// file: t_blueprint_mpi_mesh_parmetis.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include "conduit.hpp"
+#include "conduit_blueprint.hpp"
+#include "conduit_blueprint_mpi.hpp"
+#include "conduit_blueprint_mpi_mesh_parmetis.hpp"
+#include "conduit_relay.hpp"
+#include "conduit_relay_mpi.hpp"
+#include "conduit_relay_mpi_io.hpp"
+#include "conduit_relay_mpi_io_blueprint.hpp"
+#include "conduit_utils.hpp"
+#include "conduit_fmt/conduit_fmt.h"
+
+#include <mpi.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include "gtest/gtest.h"
+
+using namespace conduit;
+using namespace conduit::relay;
+using namespace conduit::relay::mpi;
+using namespace conduit::utils;
+
+using namespace std;
+//
+//
+// /*
+//     Creates an example mesh that using the following pattern:
+//
+//     |------------|------------|
+//     |            |            |
+//     |  domain 0  |            |
+//     |            |            |
+//     |------------|  domain 2..|
+//     |            |            |
+//     |  domain 1  |            |
+//     |            |            |
+//     |------------|------------|
+//
+//     The base grid i and j dims are used to size domains 0 and 1,
+//     domain 2 uses grid i, and j*2 as dims.
+//
+//     There are two topologies:
+//
+//     The `main` topology is a structured mesh of quads.
+//     The `boundary` topology is an unstructured mesh of lines,
+//     the line elements correspond to the boundary faces of main`.
+//
+//     The `main` and `boundary` topologies are defined using the same
+//     explicit coordset. Because if this we can relate the elements
+//     between them.
+//
+//     The field `ele_id` provides globally unique ids for elements
+//     of the `main` topology.
+//
+//     The field `domain_id` provides the domain number for elements
+//     of the `main` topology.
+//
+//     The field `bndry_val` is defined on the `boundary` topology
+//     as 1 for elements on the external mesh boundary, and 0 for
+//     elements on an internal mesh boundary.
+//
+//     The field `bndry_id` provides globally unique ids for elements
+//     of the `bndry` topology.
+//
+//     #######
+//     TODO
+//     #######
+//     - Add field `bndry_to_main` shows the relationship between the
+//       topologies.
+//     - Add an adj set?
+// */
```

Comment:
This looks like it's a commented-out copy of conduit_blueprint_mesh_examples_related_boundary.hpp.  remove comment?

Reply:
I need to remove this entire chunk of commented code, left it there while refactoring